packages/wallet-connectors/src/WalletConnect.ts

```diff
@@ -67,26 +71,26 @@ function isSignAndSendTransactionError(obj: any): obj is SignAndSendTransactionE
 }
 
 function accountTransactionPayloadToJson(data: AccountTransactionPayload) {
-    return JSON.stringify(data, (key, value) => {
+    return jsonUnwrapStringify(data, 'number', (key, value) => {
```

Comment:
I'm concerned about this `'number'` parameter. Does this mean that all bigints in any kinds of payloads become numbers? I don't think that's a reasonable assumption. Can't there just be a JSON encoder of `AccountTransactionPayload` that takes care of it all such that we don't have to worry about this here?

Reply:
I don't understand the difference to be honest.. we obviously want all the bigints to be numbers, as that is how they were converted prior to these changes? If we don't want this behaviour anymore we can always override the concrete conversion in the replacer function and pass undefined as the second parameter to not make any default conversion?

By the way, if a class specifies `toJSON`, it still uses this for conversion, which is why f.x. values of the `Timestamp` type from the SDK are converted using JavaScript numbers.

Reply:
We can keep it like this but just to explain my concern: It may be that with the currently supported payloads (or as least the ones we just tested), using numbers is always what we want. But that doesn't mean that it always will be going forward. I generally prefer things to be as well defined as possible. This parameter makes the JSON encoding of `AccountTransactionPayload` ill defined because it now depends on what application produced it.

If you disagree you don't have to change it.